presto-native-execution/presto_cpp/main/types/VeloxPlanConversion.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "presto_cpp/main/types/VeloxPlanValidator.h"
+#include "presto_cpp/presto_protocol/core/presto_protocol_core.h"
+#include "velox/common/memory/MemoryPool.h"
+
+namespace facebook::presto {
+
+protocol::PlanConversionResponse processVeloxPlanConversionRequest(
+    std::string planFragmentJson,
```

Comment:
you mean make the input `protocol::PlanFragment`?  The issue with that is if we parse the json before the call to `processVeloxPlanConversionRequest` I think there should be exception handling back in PrestoServer.cpp in case there is something in the PlanFragment that the converter doesn't know how to parse. Then to properly send that error response back we would end up duplicating most of what's in the function. So there wouldn't be much point of having `processVeloxPlanConversionRequest` anymore. wdyt?

Reply:
I see... We can keep this as is as well. I don't have a strong opinion about it.